src/python/grpcio/commands.py

```diff
@@ -322,6 +322,9 @@ def new_compile(obj, src, ext, cc_args, extra_postargs, pp_opts):
                 args_to_remove = c_specific_args
 
             if args_to_remove:
+                cc_args = [
```

Comment:
I may be wrong, but judging from the order of arguments in the user's command (see https://github.com/grpc/grpc/issues/40876#issuecomment-3529940897) `-std=c++17` is coming from the `extra_postargs`, not `cc_args`.